hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -121,10 +149,16 @@ public ContainerDiff diff(KeyValueContainerData thisContainer, ContainerProtos.C
   /**
    * Returns the container checksum tree file for the specified container without deserializing it.
    */
+  @VisibleForTesting
   public static File getContainerChecksumFile(ContainerData data) {
     return new File(data.getMetadataPath(), data.getContainerID() + ".tree");
   }
 
+  @VisibleForTesting
+  public static File getTmpContainerChecksumFile(ContainerData data) {
+    return new File(data.getMetadataPath(), data.getContainerID() + ".tree.tmp");
```

Comment:
Cleanup of semi-random file names is more complicated, requiring either a regex match or dedicated directory that gets wiped. By re-using the same file name any leftovers from previous runs due to a failed move are automatically overwritten by protobuf's `writeTo`. Spurious errors would cause failures in either implementation since ultimately the file write must succeed to move forwards.

I wanted to add a test for this specific case where the tmp write succeeds and the move fails, but could not get the permissions to work correctly. Removing write permissions on the destination file still allowed the move to happen.